### PR TITLE
OCMUI-4335: Add GroupsApplicationsSelector and log forwarding tree helpers

### PR DESCRIPTION
**Pull request by zherman0**
# Summary

Introduce shared GroupsApplicationsSelector (Formik + PatternFly tree) with Storybook stories, plus LogForwardingGroupTreeNode type/mock and groupSelectedLeavesByRoot helper for the chosen-panel grouping.

I had a much larger PR for this that included all the changes for this new component and the wizard changes to actually implement it.  However, it was 2k lines and 29 files.  Too much to easily review.  This is one of the larger pieces, the new GroupsApplicationSelector, so I split if off to be tested.  Since it exist by itself in this PR, we can only test the functionality and appearance of the new component with story book.

The follow on PR will test its integration into the wizard and actual API data.

Assisted by: Cursor/composer-2-fast

# Jira

Fixes [OCMUI-4335](https://issues.redhat.com/browse/OCMUI-4335)

# How to Test

1. run `yarn storybook`
2. Select the GroupsApplicationSelector and run through the examples and try it out yourself.  The basic functionality is that selecting an item from the tree view adds it to filter chips (lables) on a right hand panel.

# Screen Captures
Screenshot From 2026-04-14 18-36-49

# Review process

Please review and follow the [PR process](https://github.com/RedHatInsights/uhc-portal/blob/main/docs/pull-request-process.md).

## QE Reviewer

- [ ] _Pre-merge testing : Verified change locally in a browser (downloaded and ran code using reviewx tool)_
- [ ] Updated/created Polarion test cases which were peer QE reviewed
- [ ] Confirmed 'tc-approved' label was added by dev to the linked JIRA ticket
- [ ] (optional) Updated/created Cypress e2e tests
- [ ] Closed threads I started after the author made changes or added an explanation

[OCMUI-4335]: https://redhat.atlassian.net/browse/OCMUI-4335?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by t-hendricks — APPROVED** [review]
Tested the changes locally. Ran through the storybook and tested each story. This was very insightful. LGTM!

**Comment by dtaylor113** [comment]
Hi @zherman0, in storybook, pls combine "Docs", "No selection", "Required field",  and "with Tooltip" into just the single "Doc" story (which is by default 'no selection'), and have the component for the other stories show the required field w/ tooltip.  I can see different states see per story (disabled, partially-selected, full-group).
CleanShot 2026-04-17 at 14 12 12

**Comment by zherman0** [comment]
@dtaylor113 - I simplified the stories, let me know if that works for you

File: src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+  availableTitle?: React.ReactNode;
+  chosenTitle?: React.ReactNode;
+  isRequired?: boolean;
+  isDisabled?: boolean;
+  availableTooltip?: React.ReactNode;
+  chosenTooltip?: React.ReactNode;
+  listMinHeight?: string;
+};
+
+const buildTextById = (node: LogForwardingGroupTreeNode): Record<string, string> => {
```
Comment (dtaylor113) [inline src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx]: Why wouldn't "buildTextById()" be in logForwardingReviewHelpers.ts? Or maybe not as that is under "clusters/wizards/rosa"? Do we want a "common/GroupsApplicationsSelectorHelper.ts" if component not being used for clusters/wizards/rosa?

File: src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+  textById[node.id] = node.text;
+  if (node.children) {
+    node.children.forEach((child) => {
+      textById = { ...textById, ...buildTextById(child) };
+    });
+  }
+  return textById;
+};
+
+const getDescendantLeafIds = (node: LogForwardingGroupTreeNode): string[] => {
```
Comment (dtaylor113) [inline src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx]: Same question "getLeafDescendants()" is already in logForwardingReviewHelpers.ts, shouldn't this be in the helper class?

File: src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    return [node.id];
+  }
+  let childrenIds: string[] = [];
+  node.children.forEach((child) => {
+    childrenIds = [...childrenIds, ...getDescendantLeafIds(child)];
+  });
+  return childrenIds;
+};
+
+const getLeavesById = (node: LogForwardingGroupTreeNode): Record<string, string[]> => {
```
Comment (dtaylor113) [inline src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx]: same question about being in the helper class.

File: src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+      <FlexItem>
+        <PopoverHint hint={tooltip} />
+      </FlexItem>
+    ) : null}
+  </Flex>
+);
+
+/** PatternFly LabelGroup replaces `${remaining}` in this string when collapsing overflow labels. */
+const LABEL_GROUP_OVERFLOW_PLACEHOLDER = '{remaining}';
+const LABEL_GROUP_OVERFLOW_TEXT = `$${LABEL_GROUP_OVERFLOW_PLACEHOLDER} more`;
```
Comment (dtaylor113) [inline src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx:48]: Hi @zherman0 , can you add a storybook story which demonstrates the "more..." functionality, in the right-hand panel right.  How many apps do we need to trigger the "more..." link. -thanks

Comment (zherman0) [reply src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx:48]: I have it set for 8 currently - it works great with real API data, but I did not put in that many mock options - I will add additional options so we can see "more" on the story book.

File: src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx (hunk 1 above)
Comment (zherman0) [reply src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx]: These were all an artifact of pulling the code out of the original larger PR.  I had manually moved things around before including all the helper files.  I will go clean these up.

**Review by dtaylor113 — CHANGES_REQUESTED** [review]
Hi, looks great to me, left some comments.

File: src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx (hunk 3 above)
Comment (zherman0) [reply src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx]: Cleaned up

**Comment by dtaylor113** [comment]
Hi @zherman0, I'm concerned that logForwardingReviewHelpers.ts is in "src/components/clusters/wizards/rosa/..." directory.  Is the helpers only for rosa wizard, or would we use the same helper when displaying the component for cluster details Add/Edit modals?  If so, I think the helper class would be better placed within the same directory as the component.

File: src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+                  minHeight: listMinHeight,
+                  border: '1px solid var(--pf-v6-global--BorderColor--100)',
+                  borderRadius: 'var(--pf-v6-global--BorderRadius--medium)',
+                  overflow: 'auto',
+                }}
+              >
+                {chosenLeafIds.length === 0 ? (
+                  <EmptyState headingLevel="h4" titleText="" variant={EmptyStateVariant.sm}>
+                    <EmptyStateBody>No groups or applications selected</EmptyStateBody>
+                  </EmptyState>
```
[the remark this answers is not in the document]
Comment (zherman0) [reply src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx]: Done

File: src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+        <FlexItem flex={{ default: 'flex_1' }}>
+          <Stack hasGutter>
+            <StackItem>{paneHeading(availableTitle, isRequired, availableTooltip)}</StackItem>
+            <StackItem>
+              <SearchInput
+                value={treeFilter}
+                onChange={(_event, value) => setTreeFilter(value)}
+                onClear={() => setTreeFilter('')}
+                isDisabled={isDisabled}
+              />
```
[the remark this answers is not in the document]
Comment (zherman0) [reply src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx]: Done

File: src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+        return [item];
+      });
+
+    return mapGroupTreeToTreeViewData(treeData, treeFilter, false);
+  }, [treeData, treeFilter, chosenSet, memoizedLeavesById, memoizedNodeTexts, isDisabled]);
+
+  const selectedByGroup = useMemo(
+    () => groupSelectedLeavesByRoot(treeData, chosenLeafIds),
+    [treeData, chosenLeafIds],
+  );
```
[the remark this answers is not in the document]
Comment (zherman0) [reply src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.tsx:116]: The tree data in  both the test and the real API are very stable and almost never change.  This would be an extreme edge case.  Since the solution would be messy at best, I think keeping this simple is fine in this case.

**Review by dtaylor113 — APPROVED** [review]
looks great in storybook, code and location LGTM.

**Review by jmekkatt — CHANGES_REQUESTED** [review]
Hi @zherman0 , it looks like the tree filter breaks hierarchy for intermediate group matches. For example, If the we use a tree with 3+ levels of nesting like

1. Tree structure is Authentication > API Server Auth >Kube-apiserver
2. In the search/filter input on the left pane, type "API Server Auth"
3. Observe the tree view

Expectation is that , The full path Authentication > API Server Auth >Kube-apiserver is shown, with "API Server Auth" highlighted/expanded under "Authentication"

Actual behavior is  "Authentication" disappears. "API Server Auth" is promoted to the root level of the tree, losing its parent context. The user has no idea "API Server Auth" belongs under "Authentication."

Please review the recording

https://github.com/user-attachments/assets/2127c50a-cc2f-4b6e-97e9-ad43a09ee472

**Review by jmekkatt — COMMENTED** [review]
Hi @zherman0 , Another issue : Redundant UI for childless root nodes
When a root node has no children (e.g. Assume the "controller-manager" is a group without application ), it is treated as both a group and a application simultaneously

1. Click the checkbox for "Controller manager" in the left tree
2. Inspect the chosen panel on the right

The category  says "Controller manager" as group and the label inside also says "Controller manager"  as application (redundant).

Expectation is that, For a childless root node like controller-manager, the chosen panel should render it as a single closable label without any application.

Image

**Comment by zherman0** [comment]
> Hi @zherman0 , it looks like the tree filter breaks hierarchy for intermediate group matches. For example, If the we use a tree with 3+ levels of nesting like
> 
>     1. Tree structure is Authentication > API Server Auth >Kube-apiserver
> 
>     2. In the search/filter input on the left pane, type "API Server Auth"
> 
>     3. Observe the tree view
> 

@jmekkatt - There is no nesting in the log forwarding data.  I only designed it for 2 levels because the log_forward/groups API does not support it.  I do realize the data in the storybook can be altered, but that is not valid.  I see no requirement to handle 3+ levels.

**Comment by zherman0** [comment]
> Hi @zherman0 , Another issue : Redundant UI for childless root nodes When a root node has no children (e.g. Assume the "controller-manager" is a group without application ), it is treated as both a group and a application simultaneously
> 
>     1. Click the checkbox for "Controller manager" in the left tree
> 
>     2. Inspect the chosen panel on the right
> 

@jmekkatt - Apologies on this one.  Writing mock data allows me to do something that is not allowed.  The tree view shows groups and then the applications under the group.  You cannot have a group without an application; it would not make sense as we are assigning the application to the log forwarder.  My example here is not valid.

That said, the group name and the application name can be the same, which is how I have now updated the example.

**Comment by jmekkatt** [comment]
> > Hi @zherman0 , it looks like the tree filter breaks hierarchy for intermediate group matches. For example, If the we use a tree with 3+ levels of nesting like
> > ```
> > 1. Tree structure is Authentication > API Server Auth >Kube-apiserver
> > 
> > 2. In the search/filter input on the left pane, type "API Server Auth"
> > 
> > 3. Observe the tree view
> > ```
> 
> @jmekkatt - There is no nesting in the log forwarding data. I only designed it for 2 levels because the log_forward/groups API does not support it. I do realize the data in the storybook can be altered, but that is not valid. I see no requirement to handle 3+ levels.

Hi @zherman0 , It looks like the component is designed as a generic, reusable shared component? Additionally If the API adds 3+ levels in the future, the filter logic would silently break. I would recommend fixing it now while the code is fresh, rather than retrofitting later.

**Comment by jmekkatt** [comment]
Hi @zherman0 Also seeing some additional issues

Parent checkbox toggles hidden/filtered items
- Render GroupsApplicationsSelector with the mock data
- Type "audit" in the search/filter input
- The tree now shows only API with audit visible (and apiserver hidden by the filter)
- Click the API parent checkbox


Expected:
Only the visible leaf audit is added to the chosen panel

Actual:
Both audit and apiserver are added to the chosen panel, even though apiserver is hidden by the filter
The user unknowingly selects items they can't see.

https://github.com/user-attachments/assets/f1f6a40e-ce3b-4751-a4e5-9f5517967a74

**Comment by jmekkatt** [comment]
Hi @zherman0  , Badge count ignores active filter

Steps to reproduce :

- Render GroupsApplicationsSelector with the mock data
- Type "audit" in the search/filter input
- Look at the badge next to the API group

Expected:
Badge shows 1 (only audit is visible/matching)

Actual:
Badge shows 2 (total leaf count: audit + apiserver), even though only 1 leaf matches the filter
Screenshot 2026-04-27 at 12 19 19 PM

**Comment by jmekkatt** [comment]
Tree may not re-expand on filter change

- Render GroupsApplicationsSelector with the mock data
- Type "audit" in the search input — API group should auto-expand to show the matching leaf
- Manually collapse the API group by clicking its toggle
- Clear the filter, then type "apiserver"

Expected:
API group auto-expands again to reveal the matching leaf apiserver

Actual :
API group may remain collapsed although there is marching item.


https://github.com/user-attachments/assets/a4ed5429-09ce-471e-a673-0ff269616db8


Hi @zherman0 , Could you please check if this is a bug or working as designed?

**Comment by zherman0** [comment]
> Hi @zherman0 , It looks like the component is designed as a generic, reusable shared component? Additionally If the API adds 3+ levels in the future, the filter logic would silently break. I would recommend fixing it now while the code is fresh, rather than retrofitting later.

@jmekkatt - this is not a generic component - this is meant for log forwarding.  The API spec explicitly is designed to have only one group level with one application level. 

Trees can handle  many levels, but this component visually shows the selected items in the right panel and there would be no good way to show nested groups, even if the concept existed.  Since the API will not be changing and this is a purpose built component, I will leave it as is.

**Comment by zherman0** [comment]
> Hi @zherman0 , Badge count ignores active filter
> 
> Steps to reproduce :
> 
>     * Render GroupsApplicationsSelector with the mock data
> 
>     * Type "audit" in the search/filter input
> 
>     * Look at the badge next to the API group
> 
> 
> Expected: Badge shows 1 (only audit is visible/matching)
> 
> Actual: Badge shows 2 (total leaf count: audit + apiserver)

@jmekkatt - Per the ticket mock ups, the counters represent the items in a group, not the filtered items.  I think it would require serious discussion with UX.  

That said, I added the search as a bonus thing to help out the user, but it seems it is just causing more issues than helper at least in your opinion.  There is no acceptance criteria requiring any search, so I am removing it.  I hope that makes for a better UX.

**Comment by zherman0** [comment]
@jmekkatt - Please have another look now that I removed the filter.  It added great complexity and was not a requirement and apparently there were too many issues.  The code is now much simpler as I just show the full tree, which is what the actual ticket requests instead of me trying to go above and beyond.  Thanks.

**Comment by jmekkatt** [comment]
Hi @zherman0 , Mockups and proposed design in the ticket shows two sections (both left and right) wrapped inside a card but the implementation doesn't have these definition. Could you please take a look?
Mockup 
Screenshot 2026-04-28 at 11 02 04 AM


Actual definition
 
Screenshot 2026-04-28 at 11 03 02 AM

**Comment by jmekkatt** [comment]
> > Hi @zherman0 , It looks like the component is designed as a generic, reusable shared component? Additionally If the API adds 3+ levels in the future, the filter logic would silently break. I would recommend fixing it now while the code is fresh, rather than retrofitting later.
> 
> @jmekkatt - this is not a generic component - this is meant for log forwarding. The API spec explicitly is designed to have only one group level with one application level.
> 
> Trees can handle many levels, but this component visually shows the selected items in the right panel and there would be no good way to show nested groups, even if the concept existed. Since the API will not be changing and this is a purpose built component, I will leave it as is.

Hi @zherman0 , Could you please help me to get the list of groups and application values? I'm confused because the mockup/proposed design in the card and actual implementation seems to be different.  For example, As per the mockups, the  "Control manager" and "Ungrouped applications" groups contains couple of application inside it.

**Comment by zherman0** [comment]
@jmekkatt - Here are the actual log forwarding group values from the API:
```
{
  "kind": "LogForwardingGroupList",
  "href": "/api/clusters_mgmt/v1/log_forwarding/groups",
  "page": 1,
  "size": 5,
  "total": 5,
  "items": [
    {
      "kind": "LogForwardingGroup",
      "name": "api",
      "enabled": true,
      "versions": [
        {
          "id": "1.0",
          "applications": [
            "audit-webhook",
            "kube-apiserver",
            "oauth-openshift",
            "openshift-apiserver",
            "openshift-oauth-apiserver",
            "packageserver",
            "validation-webhook"
          ]
        }
      ]
    },
    {
      "kind": "LogForwardingGroup",
      "name": "authentication",
      "enabled": true,
      "versions": [
        {
          "id": "1.0",
          "applications": ["ignition-server", "konnectivity-agent"]
        }
      ]
    },
    {
      "kind": "LogForwardingGroup",
      "name": "controller manager",
      "enabled": true,
      "versions": [
        {
          "id": "1.0",
          "applications": [
            "aws-ebs-csi-driver-controller",
            "capi-provider-controller-manager",
            "catalog-operator",
            "cloud-controller-manager",
            "cloud-credential-operator",
            "cloud-network-config-controller",
            "cluster-network-operator",
            "cluster-node-tuning-operator",
            "cluster-policy-controller",
            "cluster-version-operator",
            "control-plane-operator",
            "control-plane-pki-operator",
            "csi-snapshot-controller-operator",
            "csi-snapshot-controller",
            "dns-operator",
            "hosted-cluster-config-operator",
            "ingress-operator",
            "kube-controller-manager",
            "machine-approver",
            "multus-admission-controller",
            "network-node-identity",
            "olm-operator",
            "openshift-controller-manager",
            "openshift-route-controller-manager",
            "ovnkube-control-plane"
          ]
        }
      ]
    },
    {
      "kind": "LogForwardingGroup",
      "name": "scheduler",
      "enabled": true,
      "versions": [{ "id": "1.0", "applications": ["kube-scheduler"] }]
    },
    {
      "kind": "LogForwardingGroup",
      "name": "autoscaling",
      "enabled": true,
      "versions": [
        { "id": "1.0", "applications": ["karpenter", "karpenter-operator"] }
      ]
    }
  ]
}
```

**Comment by jmekkatt** [comment]
Hi @zherman0 , Could you please take a look https://github.com/RedHatInsights/uhc-portal/pull/456#issuecomment-4332616689 ?

**Comment by zherman0** [comment]
> Hi @zherman0 , Could you please take a look [#456 (comment)](https://github.com/RedHatInsights/uhc-portal/pull/456#issuecomment-4332616689) ?

@jmekkatt - Apologies Jaya, I did see that but i did not get to it yesterday.  I have added a border to the cards so now the panels better match the mockup.  I also made the label bold since that is how the mockup shows it as well.

**Comment by zherman0** [comment]
@jmekkatt - Can you please review this when you get some time?  Thanks.

**Comment by jmekkatt** [comment]
Hi @zherman0 , The card sizes on the left and right sides appear misaligned compared to the mockup and expected design. It would be better to keep both cards consistent in size. (I’m not sure if this is related to the card borders.) Could you please take a look?
Screenshot 2026-05-06 at 4 14 48 PM

**Comment by jmekkatt** [comment]
> @jmekkatt - Here are the actual log forwarding group values from the API:
> 
> ```
> {
>   "kind": "LogForwardingGroupList",
>   "href": "/api/clusters_mgmt/v1/log_forwarding/groups",
>   "page": 1,
>   "size": 5,
>   "total": 5,
>   "items": [
>     {
>       "kind": "LogForwardingGroup",
>       "name": "api",
>       "enabled": true,
>       "versions": [
>         {
>           "id": "1.0",
>           "applications": [
>             "audit-webhook",
>             "kube-apiserver",
>             "oauth-openshift",
>             "openshift-apiserver",
>             "openshift-oauth-apiserver",
>             "packageserver",
>             "validation-webhook"
>           ]
>         }
>       ]
>     },
>     {
>       "kind": "LogForwardingGroup",
>       "name": "authentication",
>       "enabled": true,
>       "versions": [
>         {
>           "id": "1.0",
>           "applications": ["ignition-server", "konnectivity-agent"]
>         }
>       ]
>     },
>     {
>       "kind": "LogForwardingGroup",
>       "name": "controller manager",
>       "enabled": true,
>       "versions": [
>         {
>           "id": "1.0",
>           "applications": [
>             "aws-ebs-csi-driver-controller",
>             "capi-provider-controller-manager",
>             "catalog-operator",
>             "cloud-controller-manager",
>             "cloud-credential-operator",
>             "cloud-network-config-controller",
>             "cluster-network-operator",
>             "cluster-node-tuning-operator",
>             "cluster-policy-controller",
>             "cluster-version-operator",
>             "control-plane-operator",
>             "control-plane-pki-operator",
>             "csi-snapshot-controller-operator",
>             "csi-snapshot-controller",
>             "dns-operator",
>             "hosted-cluster-config-operator",
>             "ingress-operator",
>             "kube-controller-manager",
>             "machine-approver",
>             "multus-admission-controller",
>             "network-node-identity",
>             "olm-operator",
>             "openshift-controller-manager",
>             "openshift-route-controller-manager",
>             "ovnkube-control-plane"
>           ]
>         }
>       ]
>     },
>     {
>       "kind": "LogForwardingGroup",
>       "name": "scheduler",
>       "enabled": true,
>       "versions": [{ "id": "1.0", "applications": ["kube-scheduler"] }]
>     },
>     {
>       "kind": "LogForwardingGroup",
>       "name": "autoscaling",
>       "enabled": true,
>       "versions": [
>         { "id": "1.0", "applications": ["karpenter", "karpenter-operator"] }
>       ]
>     }
>   ]
> }
> ```

Hi @zherman0 ,I also used mock data to observe the UI behavior. When the group and application definitions are large or expanded, the card size increases significantly. It seems we may need a fixed height with a scrollbar to prevent the card from growing based on the number of applications. Could you please check this?
Kindly refer attached recordings.
https://github.com/user-attachments/assets/f3ff2d88-85bf-4093-ba41-92283c2bde29

**Comment by dtaylor113** [comment]
Hi @jmekkatt, can we consider the borders and look-n-feel of the component non-blocking and file a bug or RFE in the epic which will be fixed before we turn on this feature?  This PR is holding up 2 other PRs and we are trying to deliver this feature in the next sprint.  -thanks

**Comment by jmekkatt** [comment]
> Hi @jmekkatt, can we consider the borders and look-n-feel of the component non-blocking and file a bug or RFE in the epic which will be fixed before we turn on this feature? This PR is holding up 2 other PRs and we are trying to deliver this feature in the next sprint. -thanks

Hi @dtaylor113, just to clarify - are you suggesting that we revisit or adjust what was defined in the mockup expectations? I'm not sure if that's the right approach (and it may not qualify as an RFE), since we already have a proposed mockup that doesn’t seem fully aligned with the current behavior.
Given that this is a new UI component, it might be beneficial to address these differences early, especially if they are relatively straightforward to fix.

**Comment by dtaylor113** [comment]
Hi @jmekkatt, what I'm saying is that this 1st, initial, starting component PR has been open for 3 weeks now and I'm asking that we defer cosmetic issues such as borders and max-heights and focus on the main functionality of the component in order to unblock the 2 other dependent PRs waiting for this component.
Let's file a bug in the epic and move on.  We WILL address such issues before we flip the feature flag on, but I feel like we're getting bogged on this single component.  -thanks

**Comment by zherman0** [comment]
@jmekkatt - Fixing the heights of the card to be the same was easy enough.  However, there is a lot of CSS magic to make the scroll happen inside the cards.  I am okay with it and it does not use any PF specific versioning, but it is not straight forward.  That said, what maxHeight is the "right" height to say we should start scrolling?  For now I set it as a hardcoded 800px.  @dtaylor113 - can you please look at the latest commit and let me know what you think?
I also added a story book for showing a max height of 350px so you can see what it looks like. Thanks

**Comment by dtaylor113** [comment]
Hi @zherman0, last two commits look fine to me.  Although, I'm not crazy about custom css overriding PF defaults in src/components/common/GroupsApplicationsSelector/GroupsApplicationsSelector.scss, but this is a custom component, so I understand.

**Review by jmekkatt — APPROVED** [review]
Thanks @zherman0 for the changes.
Verification of the group application selector has done. LGTM
